test/SlowTests/Server/Documents/Expiration/ExpirationTests.cs

```diff
@@ -401,8 +401,10 @@ public async Task ShouldDeleteExpiredDocumentsForSharding(Options options)
                         foreach (var database in kvp.Value)
                         {
                             database.Time.UtcDateTime = () => DateTime.UtcNow.AddMinutes(10);
-                            var expiredDocumentsCleaner = database.ExpiredDocumentsCleaner;
-                            await expiredDocumentsCleaner?.CleanupExpiredDocs()!;
+                            if (database.ExpiredDocumentsCleaner == null)
+                                continue;
```

Comment:
when is this possible? you are setting the expiration so it should not be null?

Reply:
I think it would be better to use WaitForValue, yes

Reply:
done